Assets/Scripts/SS3D/Systems/Construction/ConstructionHelper.cs

```diff
@@ -0,0 +1,104 @@
+using SS3D.Core.Behaviours;
+using SS3D.Systems.Tile;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace SS3D.Systems.Construction.UI
+{
+    public class ConstructionHelper: Actor
+    {
+        public Material validConstruction;
+        public Material invalidConstruction;
+        public Material deleteConstruction;
+
+        private GameObject _ghostObject;
+        private Vector3 _targetPosition;
+        private Direction _dir = Direction.North;
+
+        public enum BuildMatMode
+        {
+            Valid,
+            Invalid,
+            Building,
+            Deleting
+        }
+
+        public void CreateGhost(GameObject prefab)
+        {
+            if (_ghostObject == null)
+            {
+                _ghostObject = Instantiate(prefab);
+
+                var colliders = _ghostObject.GetComponents<Collider>();
+                foreach (Collider col in colliders)
+                {
+                    col.enabled = false;
+                }
+            }
+        }
+        public void DestroyGhost()
+        {
+            if (_ghostObject != null)
+            {
+                DestroyImmediate(_ghostObject);
+                _ghostObject = null;
+            }
+        }
+
+        public void SetTargetPosition(Vector3 target)
+        {
+            _targetPosition = target + new Vector3(0, 0.2f, 0);
+        }
+
+        public void MoveGhost()
+        {
+            // Required if the object has a network script attached
+            _ghostObject.SetActive(true);
+
+            _ghostObject.transform.position = Vector3.Lerp(_ghostObject.transform.position, _targetPosition, Time.deltaTime * 15f);
+            _ghostObject.transform.rotation = Quaternion.Lerp(_ghostObject.transform.rotation, Quaternion.Euler(0, TileHelper.GetRotationAngle(_dir), 0), Time.deltaTime * 15f);
+        }
+
+        public void ChangeGhostColor(BuildMatMode mode)
+        {
+            Material ghostMat = null;
+
+            switch (mode)
```

Comment:
make a small comment like "choose color appropriately for the ghost, depending on it's state" or something